sparse_strips/vello_api/src/paint.rs

```diff
@@ -99,6 +99,45 @@ pub struct Image {
     pub transform: Affine,
 }
 
+/// A premultiplied color.
+#[derive(Debug, Clone, PartialEq, Copy)]
+pub struct PremulColor {
+    premul_u8: PremulRgba8,
+    premul_f32: peniko::color::PremulColor<Srgb>,
+}
+
+impl PremulColor {
+    /// Create a new premultiplied color.
+    pub fn new(color: AlphaColor<Srgb>) -> Self {
+        let premul = color.premultiply();
+
+        Self {
+            premul_u8: premul.to_rgba8(),
+            premul_f32: premul,
+        }
+    }
+
+    /// Return the color as a premultiplied RGBA8 color.
+    pub fn rbga_u8(&self) -> [u8; 4] {
+        self.premul_u8.to_u8_array()
+    }
+
+    /// Return the color as a premultiplied RGBA8 color, packed into a u32 in little-endian order.
+    pub fn rbga_u32(&self) -> u32 {
+        self.premul_u8.to_u32()
+    }
+
+    /// Return the color as a premultiplied RGBA32 color in the range [0.0, 1.0].
+    pub fn rbga_f32(&self) -> [f32; 4] {
+        self.premul_f32.components
+    }
```

Comment:
Would it make sense to give access to the underlying color values, by making the `PremulColor` fields `pub` or having the following?

```rust
pub fn as_premul_rgba8(&self) -> PremulRgba8;
pub fn as_premul_f32(&self) -> PremulColor<Srgb>
```

That way the normal `color` APIs can be used by consumers.